src/users.ts

```diff
@@ -4,76 +4,77 @@ import { APIClient } from '@textile/users-grpc/users_pb_service'
 import { ServiceError } from '@textile/hub-grpc/hub_pb_service'
 import { Context } from '@textile/context'
 import { ThreadID } from '@textile/threads-id'
+import { Client } from '@textile/threads-client'
 
 const logger = log.getLogger('users')
 
-/**
- * Buckets is a web-gRPC wrapper client for communicating with the web-gRPC enabled Textile Buckets API.
- */
-export class Users {
-  private client: APIClient
-
-  /**
-   * Creates a new gRPC client instance for accessing the Textile Buckets API.
-   * @param context The context to use for interacting with the APIs. Can be modified later.
-   */
-  constructor(public context: Context = new Context()) {
-    this.client = new APIClient(context.host, {
-      transport: context.transport,
-      debug: context.debug,
-    })
+// Module augmentation to add methods to Peer from core
```

Comment:
This is pretty cool, we essentially extend the Client interface on the fly here, and then the implementations follow